### DISCUSSION TITLE
Un error del MVN

### DISCUSSION THREAD

**Discussion by jfloresmaquera** [dds-jv/foro#227] [Consultas de herramientas] [answered]
Estabamos haciendo el TPA y cuando hicimos el mvn verify nos salto multiples errores todos del mismo tipo:
 URF_UNREAD_FIELD
Por lo que pudimos leer y hacer una rapida busqueda es porque tenemos muchas variables que creamos y guardamos pero que nunca leemos o usamos.
¿Interpretamos correctamente el error?¿Habría alguna forma de poder instanciar este error de forma que no salte en el mvn verify?

**Comment by nanderson87** [comment, marked as answer]
Hola @jfloresmaquera,

El  `URF_UNREAD_FIELD` básicamente es algo como lo que comentaste vos. Para ser un poco más preciso son atributos definidos en una clase que pueden estar (o no) inicializados con algún valor pero que finalmente nunca son utilizados. Es decir que esa misma clase no lo utiliza en ninguno de sus métodos ni otro objeto esta interactuando con ellos (en algún caso particular en donde querian que el atributo sea publico por algún motivo)

Te consulto para entender. En este momento en qué los estaría trabando el `mvn verify` para el desarrollo del TPA?

**Reply by jfloresmaquera** [reply]
Exactamente, estabamos arreglando el TPA después de la revision del jueves pasado y nos saltaba ese error con el mvn verify.

**Reply by flbulgarelli** [reply]
¡Hola!

El error es correcto, y como bien vos y @nanderson87 dicen, se debe a que tenés estado que nunca usás. Entonces deberías preguntarte: 

1. Si nunca se usa, ¿para qué se agrega?
2. Si aún así consideran que lo necesitan porque charlaron que es información descriptiva que más adelante van a consultar (por ejemplo, el teléfono de contacto o su dirección) entonces la forma de indicarle a Java (y al resto de las personas que programan el sistema) que algo está pensado para ser consultado es ponerle un getter. De esa forma, deja de ser tan solo parte del estado interno del objeto y pasa a ser parte de la interfaz del mismo. 

¡Saludos!

**Reply by gastonprieto** [reply]
Te sirvio @jfloresmaquera ?

**Reply by jfloresmaquera** [reply]
Si si, me sirvio, gracias por la respuesta. Pense que habia escrito para confirmar.
